src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.FeatureCollection.cs

```diff
@@ -276,6 +279,17 @@ async Task<Stream> IHttpUpgradeFeature.UpgradeAsync()
             return new DuplexStream(RequestBody, ResponseBody);
         }
 
+        Task<X509Certificate2> ITlsConnectionFeature.GetClientCertificateAsync(CancellationToken cancellationToken)
+        {
+            return Task.FromResult(ClientCertificate);
+        }
+
+        unsafe X509Certificate2 ITlsConnectionFeature.ClientCertificate
+        {
+            get => ClientCertificate;
```

Comment:
Why have this and separate ClientCertificate property?

Reply:
Pattern that all other features follow, like RequestHeaders, Request/Response Body, etc.: https://github.com/aspnet/IISIntegration/blob/release/2.2/src/Microsoft.AspNetCore.Server.IIS/Core/IISHttpContext.FeatureCollection.cs#L105-L109. Main reason is to prevent app code from overwriting critical features.

Reply:
I don't like it in other places too. How does it prevent overwriting?

Reply:
If someone does ` context.Features.Get<IHttpRequestFeature>.Body` and replaces the body, bad things will happen with the request/repsonse. 

cc/ @Tratcher as this was copied from HttpSys.

Reply:
But that's the reason it's writable and can see scenarios where ClientCertificate can be replaced.

Reply:
There are some fields where the server needs to keep a reference to the original value like headers or body, and those must be preserved in and referenced from a private field. ClientCertificate is not a field that the server has any need for (unless it needs to dispose it?).

Reply:
Doesn't look like we need to dispose it.